src/Components/Blazor/WebAssembly.Authentication/test/RemoteAuthenticationServiceTests.cs

```diff
@@ -0,0 +1,492 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text.Json;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Options;
+using Microsoft.JSInterop;
+using Xunit;
+
+namespace Microsoft.AspNetCore.Components.WebAssembly.Authentication
+{
+    public class RemoteAuthenticationServiceTests
+    {
+        [Fact]
+        public async Task RemoteAuthenticationService_SignIn_UpdatesUserOnSuccess()
+        {
+            // Arrange
+            var testJsRuntime = new TestJsRuntime();
+            var options = CreateOptions();
+            var runtime = new RemoteAuthenticationService<RemoteAuthenticationState, OidcProviderOptions>(
+                testJsRuntime,
+                options);
+
+            var state = new RemoteAuthenticationState();
+            testJsRuntime.SignInResult = new RemoteAuthenticationResult<RemoteAuthenticationState>
+            {
+                State = state,
+                Status = RemoteAuthenticationStatus.Success
+            };
+
+            // Act
+            await runtime.SignInAsync(new RemoteAuthenticationContext<RemoteAuthenticationState> { State = state });
+
+            // Assert
+            Assert.Equal(
+                new[] { "AuthenticationService.init", "AuthenticationService.signIn", "AuthenticationService.getUser" },
+                testJsRuntime.PastInvocations.Select(i => i.identifier).ToArray());
+        }
+
+        [Theory]
+        [InlineData(RemoteAuthenticationStatus.Redirect)]
+        [InlineData(RemoteAuthenticationStatus.Failure)]
+        [InlineData(RemoteAuthenticationStatus.OperationCompleted)]
+        public async Task RemoteAuthenticationService_SignIn_DoesNotUpdateUserOnOtherResult(string value)
+        {
+            // Arrange
+            var testJsRuntime = new TestJsRuntime();
+            var options = CreateOptions();
+            var runtime = new RemoteAuthenticationService<RemoteAuthenticationState, OidcProviderOptions>(
+                testJsRuntime,
+                options);
+
+            var state = new RemoteAuthenticationState();
+            testJsRuntime.SignInResult = new RemoteAuthenticationResult<RemoteAuthenticationState>
+            {
+                Status = value
+            };
+
+            // Act
+            await runtime.SignInAsync(new RemoteAuthenticationContext<RemoteAuthenticationState> { State = state });
+
+            // Assert
+            Assert.Equal(
+                new[] { "AuthenticationService.init", "AuthenticationService.signIn" },
+                testJsRuntime.PastInvocations.Select(i => i.identifier).ToArray());
```

Comment:
Here too, can we say anything about the authentication state on the .NET side?

Reply:
Seems it applies to lots of tests here so I'll skip posting the same comment elsewhere :)

Reply:
Same as https://github.com/dotnet/aspnetcore/pull/18851/files#r379475764 the lack of `AuthenticationService.getUser` means that we didn't try to update the user.